src/QirRuntime/test/QIR-static/qir-test-math.cpp

```diff
@@ -23,3 +27,52 @@ TEST_CASE("QIR: Math.ArcTan2", "[qir.math][qir.Math.ArcTan2]")
     REQUIRE(0 == Microsoft__Quantum__Testing__QIR__Math__ArcTan2Test__body());
 }
 
+TEST_CASE("QIR: Math.DrawRandomInt", "[qir.math][qir.Math.DrawRandomInt]")
+{
+    // Range of the generated random numbers:
+    constexpr int64_t   minimum = -1024;
+    constexpr int64_t   maximum =  1023;
+    constexpr size_t    count =  maximum - minimum + 1;     // How many numbers are in the range.
+
+    size_t genAverage = 1000;                               // Each number should be generated ~1000 times.
+    size_t times = count * genAverage;                      // How many times to draw.
+
+    std::vector<uint16_t>  counters(count, 0);              // How many times each random number has been generated.
+
+    // Count how many times each random number has been generated.
+    while(--times)
+    {
+        const int64_t randomNumber = 
+            Microsoft__Quantum__Testing__QIR__Math__DrawRandomIntTest__body(minimum, maximum);
+        REQUIRE(minimum <= randomNumber);
+        REQUIRE(randomNumber <= maximum);
+
+        // Increment the counter for this number:
+        const int64_t index = randomNumber - minimum;
+        REQUIRE(0 <= index);
+        REQUIRE(index < counters.size());
+        ++(counters[index]);
+    }
+
+    // Make sure that each number has been generated, and the distribution is approximately uniform:
+    for(auto counter : counters)
```

Comment:
;-) No, I don't know with what probability this is expected to fail, really. I was studying the probability theory about 25 years ago, and math was not my major (SW dev was the major). I'll take a deeper look into your formulas after I commit fixes (and get the build process running for this PR), and am done with Irina's CR.  
If you have good links for me to look at (in order to refresh my memory in this particular part of probability) then please feel free to provide.

Reply:
I think we should avoid adding any tests that have a nonzero probability of failure. To test random distributions, it would make more sense to me to use a fixed random seed. Even if the probability of failure is very low, we run a lot of CI builds, and having nondeterministic CI failures is not fun. :)

Reply:
Oh... Thank you for the insight, @samarsha.
As I understand, at least for the test purposes, the sequence of the numbers generated by the DrawRandomInt() should be 100% predictable (fully controlled/customized pseudo-random). And the test should compare the actual sequence with the expected one.
Otherwise, the intermittent test run failures will be bothering the CI/PR builds/runs for the unrelated changes.
I will push my current changes (to make sure the build is fixed and earlier feedback is addressed), and then will fix this test. For now I will comment out the logic that can fail intermittently.

Reply:
I am confused, are you saying that repeated runs of a Q# program that calls DrawRandomInt() would produce the same sequence of integers?

Reply:
If the random seed is fixed, then the sequence from DrawRandomInt should be the same every time.

Reply:
There is no means on Q#'s side to provide the seed, right? So, the Q# code will always generated the same sequence, right? Is that the expected behaviour?

Reply:
No, I don't think Q# has any way to set the seed. The default behavior is to randomize the seed, so each program execution is different. I'm suggesting that we should fix the seed only for tests, so that tests that rely on randomness pass deterministically.

Reply:
In general, there may not even be a seed. E.g.: if a QRNG is used to back `DrawRandomInt()`, there's no pseudorandom RNG to seed in that case.

Reply:
That's true, so I don't think a QRNG should be used as a source of randomness for CI tests.

Reply:
Very much so agreed! Was only speaking to why it's intentional that there's no Q# API for setting the seed.

Reply:
As for probability of failure, a back of the envelope calculation puts that at around 10⁻²² for the number of samples that are taken so far, such that we would need around 10¹¹ builds to have any nontrivial support over the number of failures being at least one. We may still prefer deterministic for all the reasons discussed above, but I think the rate of failure from randomness in this test is comparable to the rate of failure caused by cosmic rays hitting build agent RAM.

Reply:
I would say, ideally I need to make the random number generator customizable, such that by default the generated random numbers are as random as possible, but _for the test purposes_ we could change the settings minimally, such that the generated random numbers are 100% predictable, and as much as possible of the functionality to be tested.
To summarize, I will be targeting at different random number sequences _by default_ and _during the test_. 
I'll get back to this.

Reply:
@kuzminrobin: I don't disagree with customizability, but I think it is critical for that customization to **not** be exposed at the Q# level. In particular, Q# as a language and as a platform is designed to be transparent across devices (including simulators), so that a Q# developer has confidence that their quantum applications are not inadvertently specialized to a particular simulator or device — this is important enough that it's even our first [language design principle](https://github.com/microsoft/qsharp-language/#design-principles).

From that perspective, a developer looking to control how a particular target provides random numbers should do so in a way that makes it explicit that they're configuring the target and not at the Q# level.

Reply:
Before I posted my previous message I didn't refresh the page for a number of hours, and didn't notice a number of messages ;-). I was basically answering the Irina's message 
> I am confused, are you saying that repeated runs of a Q# program that calls DrawRandomInt() would produce the same sequence of integers?

Before I continue working on this I'll push the changes addressing the other CR notes.

Reply:
I have updated my branch, please review the changes relating to this discussion.